modules/RoutingContext.js

```diff
@@ -5,15 +6,28 @@ import getRouteParams from './getRouteParams'
 
 const { array, func, object } = React.PropTypes
 
-/**
- * A <RoutingContext> renders the component tree for a given router state
- * and sets the history object and the current location in context.
- */
 class RoutingContext extends Component {
 
   getChildContext() {
     const { history, location } = this.props
-    return { history, location }
+    const router = {
```

Comment:
Micro-optimization-ville, population me: it might be good to put this in `componentWillReceiveProps` after a reference equality check, to avoid re-building the `router` object unnecessarily.